howtos/diffs/ensembling.diff

```diff
@@ -94,26 +94,43 @@ index 51d2fde..0ca3365 100644
 +  test_ds = jax_utils.replicate(test_ds)
  
    for epoch in range(1, num_epochs + 1):
-     optimizer, _ = train_epoch(
+     optimizer, train_metrics = train_epoch(
          optimizer, train_ds, batch_size, epoch, input_rng)
      loss, accuracy = eval_model(optimizer.target, test_ds)
 -    logging.info('eval epoch: %d, loss: %.4f, accuracy: %.2f',
 +    # `loss` and `accuracy` are now 1D arrays of length `jax.device_count()`
 +    logging.info('eval epoch: %d, loss: %s, accuracy: %s',
                   epoch, loss, accuracy * 100)
-   return optimizer
- 
+     summary_writer.scalar('train_loss', train_metrics['loss'], epoch)
```

Comment:
I know this isn't a problem in this PR, but I have confused myself now -- what is the "train loss" scalar for an ensemble training?

I'm just putting it here to share my confusing but it shouldn't block merging this PR.